src/impy/models/dpmjetIII.py

```diff
@@ -81,6 +82,19 @@ def __init__(self, event_kinematics, seed=None, logfname=None):
         dpm_conf = impy_config["dpmjetIII"]
 
         info(1, "Initializing DPMJET-III")
+
+        dpmjet_ver = pathlib.Path(dpm_conf["dat_dir"][self.version]).parts[-1]
+        if dpmjet_ver == "dpm3":
+            _cached_data_dir(
```

Comment:
_cached_data_dir should return the data directory for the model given a key from the config, that the model pixks up. The model should not care in which directory exactly the data is stored.
Then it should generate the download link by itself. https://github.. should be in config.

Reply:
All data regarding paths and urls are now in `config.py` file.  `_cached_data_dir` returns a string (not Path) in format: `/full/path/to/data/dir/", as all models require string. The last slash is critical.

Reply:
We don't need to configure this. The logic that computes the path can be in `_cached_data_dir`. The rule is not complicated. Generate a local directory with the same name as the last part of the remote url inside `impy/iamdata`.

Reply:
It was like this, except for version. Maybe I haven't understood correctly, but this is the design which we discussed with @afedynitch.

Reply:
Then let's discuss it publically, because I don't know what you discussed with Anatoli. I don't want to have a `config.py` file at all. Once this is done, I will work on moving all static config code into the respective models submodules, so that it sits next to the interface code. This is increasing encapsulation and locality of code that belongs together.

Settings that we want to change dynamically should especially not go into a config.py. We call methods on the model class to change behavior or pass options via the `init` if necessary. All dynamic configuration must live in instance variables so that two instances can have different configurations. Currently all models except Pythia8 cannot have two instances in parallel anway, so the difference between instance variables and global variables is mute for those, but Pythia8 exists, too, and therefore the API cannot be designed upfront in this way.

Reply:
I think it is easy to change, when you will move configs to the submodules. Actually, `_cached_data_dir` accepts in the end only names of directories `dpm3`, `dpm3191`, `qgsjet`, `epos`, `Pythia8`. They are just taken from config currently. I can move 

```python
data_url = "https://github.com/impy-project/impy/releases/download/zipped_data_v1.0"
data_version = "001"
data_path = Path(__file__).parent.absolute() / "iamdata"
```

to `_cached_data_dir`. And hardcode (or take from the variable in the submodule?) `dpm3`, `dpm3191`... and file names taken from config. 

I can do that , but I would like to be sure that this is what we want in the end.

Reply:
Yes, what I had in mind was something like this:

```py
class EposLHC:
     def __init__(self, ...):
           _data_dir = _cached_data_dir("https://github.com/impy-project/impy/releases/download/eposlhc.zip")
           # creates local path Path(impy.__file__) / "iamdata" / eposlhc and `_data_dir` is the string to that path
```

Calling _cached_data_dir inside `__init__` is important so that the download is delayed until the point when the user actually creates an instance of the class. If `_data_dir` was a class variable, the download would start already when we import this submodule, which we always do for all models if I remember correctly.